controllers/azuremachine_reconciler.go

```diff
@@ -73,6 +73,9 @@ func (s *azureMachineService) Reconcile(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.azureMachineService.Reconcile")
 	defer done()
 
+	// Ensure that the deprecated networking field values have been migrated to the new NetworkInterfaces field.
+	s.scope.AzureMachine.Spec.SetNetworkInterfacesDefaults()
+
```

Comment:
AzureMachine Defaulter should take care of this. I don't think we need this here.

Reply:
It is possible that the reconciler runs before a resource has been modified and therefore before the defaulter has run. This is a protection against that case. The code assumes those new fields are populated and we will cause the reconciler to panic with nil pointer references if a resource has not yet had these fields set.

Reply:
I don't think this code base follows "setting defaults in the reconciler" convention. @CecileRobertMichon What do you say?

Reply:
@nawazkh the code already does this (predating this PR) here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azuremachine_reconciler.go#L76-L78

Reply:
Agree, we have to do this to preserve backward compatibility